amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/QueueContainer.java

```diff
@@ -15,21 +15,43 @@
 package io.streamnative.pulsar.handlers.amqp;
 
 import com.google.common.collect.Maps;
+import io.streamnative.pulsar.handlers.amqp.impl.PersistentQueue;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.service.Topic;
+import org.apache.pulsar.broker.service.persistent.PersistentTopic;
 import org.apache.pulsar.common.naming.NamespaceName;
 
 /**
  * Container for all queues in the broker.
  */
+@Slf4j
 public class QueueContainer {
+    private Executor executor = Executors.newCachedThreadPool();
+
+    private AmqpTopicManager amqpTopicManager;
+    private PulsarService pulsarService;
+    private ExchangeContainer exchangeContainer;
+
+    protected QueueContainer(AmqpTopicManager amqpTopicManager, PulsarService pulsarService,
+                             ExchangeContainer exchangeContainer) {
+        this.amqpTopicManager = amqpTopicManager;
+        this.pulsarService = pulsarService;
+        this.exchangeContainer = exchangeContainer;
+    }
 
     @Getter
-    private static Map<NamespaceName, Map<String, AmqpQueue>> queueMap = new ConcurrentHashMap<>();
+    private Map<NamespaceName, Map<String, AmqpQueue>> queueMap = new ConcurrentHashMap<>();
 
-    public static void putQueue(NamespaceName namespaceName, String queueName, AmqpQueue amqpQueue) {
+    public void putQueue(NamespaceName namespaceName, String queueName, AmqpQueue amqpQueue) {
```

Comment:
This is a `void` method.  But, on line 61 you are returning a map.  Which one is "right"?

Reply:
The return map is for the inner function, not for the outter putQueue.